src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpLogExporter.cs

```diff
@@ -28,7 +28,7 @@ namespace OpenTelemetry.Exporter;
 /// Exporter consuming <see cref="LogRecord"/> and exporting the data using
 /// the OpenTelemetry protocol (OTLP).
 /// </summary>
-internal sealed class OtlpLogExporter : BaseExporter<LogRecord>
+public sealed class OtlpLogExporter : BaseExporter<LogRecord>
```

Comment:
@CodeBlanch How do you feel about using a `public sealed` modifier instead of just `public`? All the other public classes are not marked `sealed`. Should we unmark `sealed` from this class as well?

Reply:
Probably fine to make it `public`. The reason to seal it would be to reserve the right to make changes to it in the future by preventing people from making more derived types. But just looking at the class, I don't see a strong need for that.

If we really wanted to make it nicely extensible we should consider the use cases. The class as it is probably isn't too useful for extension. Consider something like this...

```csharp
public class OtlpLogExporter
{
    public override ExportResult Export(in Batch<LogRecord> logRecordBatch)
    {
            // ...some logic omitted...
            request.AddBatch(this.sdkLimitOptions, this.ProcessResource, activityBatch);

            OnRequestSending(request); // Extension point

            if (!this.exportClient.SendExportRequest(request))
            {
                return ExportResult.Failure;
            }
            // ...some logic omitted...
    }

    // Give a nice extension point for manipulating the request before it is sent
    protected virtual void OnRequestSending(OtlpCollector.ExportLogsServiceRequest request) {}
}
```

Not saying we need to do this, just food for thought 😄

Reply:
Could be a good addition in future when we sort out the `OtlpCollector.ExportLogsServiceRequest` dependency. Right now the types are not exposed.